### DISCUSSION TITLE
Limitations of using cartesian basis functions?

### DISCUSSION THREAD

**Discussion by hatemhelal** [merzlab/QUICK#383] [Q&A]
I am curious if anyone can provide some guidance on the limitations of using the cartesian variant of Gaussian basis functions rather than the spherical harmonics based variant.  My own understanding that there are linear dependencies between the cartesian basis functions once you go beyond the p-orbitals and this can pose problems for convergence.  Has this been observed in QUICK?

**Comment by agoetz** [comment]
Linear dependence is usually not an issue.

For angular momenta `l>1` the number of Cartesian functions, `(l+1)(l+2)/2`, is greater than the number of spherical harmonics, `2l+1`. The conversion of Cartesian to spherical harmonics projects out linear combinations of Cartesian functions that have effectively lower angular momentum. For example, 6 Cartesian d functions yield 5 spherical harmonic d functions and one radially symmetric or s function. This s function is discarded. If the basis set also has an s function with similar exponent, then this can lead to linear dependences when using Cartesian functions. Otherwise, it is not an issue. 

While some basis sets were designed to be used with Cartesian functions, many basis sets were designed to be used with spherical harmonics. In this case, when using Cartesian functions, the basis set is overcomplete (more complete), which means that energies will be lower. This difference will be small for relative energies / energy differences. But you have to keep this in mind when comparing absolute energies obtained from calculations that used spherical harmonics vs Cartesian funcionts.
